code/ui/components/src/tooltip/WithTooltip.tsx

```diff
@@ -4,98 +4,125 @@ import ReactDOM from 'react-dom';
 import { styled } from '@storybook/theming';
 import { global } from '@storybook/global';
 
-import type { TriggerType } from 'react-popper-tooltip';
+import type { Config as ReactPopperTooltipConfig, PopperOptions } from 'react-popper-tooltip';
 import { usePopperTooltip } from 'react-popper-tooltip';
-import type { Modifier, Placement } from '@popperjs/core';
 import { Tooltip } from './Tooltip';
 
 const { document } = global;
 
 // A target that doesn't speak popper
-const TargetContainer = styled.div<{ mode: string }>`
+const TargetContainer = styled.div<{ trigger: ReactPopperTooltipConfig['trigger'] }>`
   display: inline-block;
-  cursor: ${(props) => (props.mode === 'hover' ? 'default' : 'pointer')};
+  cursor: ${(props) =>
+    props.trigger === 'hover' || props.trigger.includes('hover') ? 'default' : 'pointer'};
 `;
 
-const TargetSvgContainer = styled.g<{ mode: string }>`
-  cursor: ${(props) => (props.mode === 'hover' ? 'default' : 'pointer')};
+const TargetSvgContainer = styled.g<{ trigger: ReactPopperTooltipConfig['trigger'] }>`
+  cursor: ${(props) =>
+    props.trigger === 'hover' || props.trigger.includes('hover') ? 'default' : 'pointer'};
 `;
 
 interface WithHideFn {
   onHide: () => void;
 }
 
-export interface WithTooltipPureProps {
+export interface WithTooltipPureProps
+  extends Omit<ReactPopperTooltipConfig, 'closeOnOutsideClick'>,
+    PopperOptions {
   svg?: boolean;
-  trigger?: TriggerType;
-  closeOnClick?: boolean;
-  placement?: Placement;
-  modifiers?: Array<Partial<Modifier<string, {}>>>;
+  withArrows?: boolean;
   hasChrome?: boolean;
   tooltip: ReactNode | ((p: WithHideFn) => ReactNode);
   children: ReactNode;
-  tooltipShown?: boolean;
-  onVisibilityChange?: (visibility: boolean) => void | boolean;
```

Comment:
These changes seem to be breaking changes, which I'm not sure what the process when it comes to making them in storybook components (which are used by many other packages).

@kylegach could you weigh in?

Reply:
Dang. You aren't kidding: https://github.com/search?q=%22@storybook/components%22+language:JSON&type=code&l=JSON

I'm not sure these are breaking changes, though. I think the removed props are all part of the extended interfaces.

If they _are_ breaking changes, then I'm not sure why the process wouldn't be to follow semver and release a new major?

Reply:
@yannbf I have addressed your valid concerns in https://github.com/storybookjs/storybook/pull/20783/commits/875f9caccb4782ad3e73cdd400f9ab71c50e37e8 Please take another look.